### PR TITLE
[Refactor] Migrate all command-related interface functions from ProcessServiceImpl

### LINKED ISSUE #12468: [Refactor] Migrate all command-related interface functions from ProcessServiceImpl
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

+ This is a part of #12403 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by DarkAssassinator**
## Purpose of the pull request

+ This PR close #12468 
+ This is a part of #12403 

## Brief change log

+ Migrate command-related interface functions to `CommandService`
+ Add `ParamUtils`

## Verify this pull request
This pull request is already covered by existing tests

### CODE REVIEW

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/command/CommandService.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+     */
+    boolean verifyIsNeedCreateCommand(Command command);
+
+    /**
+     * insert or update command
+     *
+     * @param command command
+     * @return save command result
+     */
+    int saveCommand(Command command);
```
Comment (ruanwenjun) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/command/CommandService.java]:
```suggestion
    int upsertCommand(@Nonnull Command command);
```

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/command/CommandService.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+     * @param message message
+     */
+    void moveToErrorCommand(Command command, String message);
+
+    /**
+     * Create new command
+     * @param command command
+     * @return result
+     */
+    int createCommand(Command command);
```
Comment (ruanwenjun) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/command/CommandService.java:45]: This method make confusion with `saveCommand`, we would better don't keep both.

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/command/CommandService.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    /**
+     * Get command page
+     * @param pageSize page size
+     * @param pageNumber page number
+     * @return command page
+     */
+    List<Command> findCommandPage(int pageSize, int pageNumber);
```
Comment (ruanwenjun) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/command/CommandService.java]: Dose this method still used?

Comment (DarkAssassinator) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/command/CommandService.java]: > Dose this method still used?

i think not, i will remove it to keep code clean.

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/command/CommandService.java (hunk 2 above)
Comment (DarkAssassinator) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/command/CommandService.java:45]: > This method make confusion with `saveCommand`, we would better don't keep both.

+1, I think `saveCommand` should be made a private method, not a interface function, i will improve it

**Review by ruanwenjun — APPROVED** [review]

**Pull request merged**